machines/vic/install.go

```diff
@@ -0,0 +1,309 @@
+	executor *reification.Dispatcher
+}
+
+var (
+	BUILD_ID string
+	data     = &Data{}
+	images   = map[string][]string{
+		"core":  []string{"appliance.iso"},
+		"linux": []string{"bootstrap.iso"},
+	}
+)
+
+func init() {
+	flag.StringVar(&data.target, "target", "", "ESXi or vCenter FQDN or IPv4 address")
+	flag.StringVar(&data.user, "user", "", "ESX or vCenter user")
+	flag.StringVar(&data.passwd, "passwd", "", "ESX or vCenter password")
+	flag.StringVar(&data.cert, "cert", "", "Virtual Container Host x509 certificate file")
+	flag.StringVar(&data.key, "key", "", "Virtual Container Host private key file")
+	flag.StringVar(&data.computeResource, "pool", "", "Compute resource path, e.g. /ha-datacenter/host/myCluster/Resources/myRP")
+	flag.StringVar(&data.imageDatastore, "iStore", "", " Image datastore name")
+	flag.StringVar(&data.containerDatastore, "cStore", "", " Container datastore name - defaults to image datastore")
+	flag.StringVar(&data.displayName, "name", "docker-appliance", "The name of the Virtual Container Host")
+	flag.StringVar(&data.externalNetwork, "extNet", "", "The external network (can see hub.docker.com)")
+	flag.StringVar(&data.managementNetwork, "mgmtNet", "", "The management network (can see target)")
+	flag.StringVar(&data.bridgeNetwork, "bdgNet", "", "The bridge network")
+	flag.StringVar(&data.applianceISO, "appIso", "", "The appliance iso")
+	flag.StringVar(&data.bootstrapISO, "bootIso", "", "The bootstrap iso")
+	flag.BoolVar(&data.force, "force", false, "Force the install, removing existing if present")
+	flag.BoolVar(&data.tlsGenerate, "gen", false, "Generate certificate for Virtual Container Host")
+
+	flag.Parse()
+}
+
+func processParams() {
+	flag.Usage = func() {
+		fmt.Fprintf(os.Stderr, "%s BUILD ID: %s\n", os.Args[0], BUILD_ID)
+		flag.PrintDefaults()
+		os.Exit(1)
+	}
+
+	if data.target == "" {
+		log.Errorf("Target argument must be specified")
+		flag.Usage()
+	}
+
+	if data.user == "" {
+		log.Errorf("-user User to login target must be specified")
+		flag.Usage()
+	}
+
+	if data.computeResource == "" {
+		log.Errorf("-pool Compute resource path must be specified")
+		flag.Usage()
+	}
+
+	if data.imageDatastore == "" {
+		log.Errorf("-iStore Image datastore name must be specified")
+		flag.Usage()
+	}
+
+	if data.cert != "" && data.key == "" {
+		log.Errorf("key cert should be specified at the same time")
+	}
+	if data.cert == "" && data.key != "" {
+		log.Errorf("key cert should be specified at the same time")
+	}
+
+	if data.externalNetwork == "" {
+		data.externalNetwork = "VM Network"
+	}
+
+	if data.bridgeNetwork == "" {
+		data.bridgeNetwork = data.displayName
+	}
+
+	if len(data.displayName) > MaxDisplayNameLen {
+		log.Fatalf("Display name %s exceeds the permitted 31 characters limit. Please use a shorter -name parameter", data.displayName)
+	}
+
+	data.osType = "linux"
+	data.timeout = 3 * time.Minute
+	data.logfile = "install.log"
+
+	data.conf = config.NewConfig()
+	data.conf.NumCPUs = 1
```

Comment:
Please add a // FIXME comment for all those hardcoded values throughout this file so that we can go back and fix them.